test/lit/passes/stack-ir-defaults.wast

```diff
@@ -0,0 +1,66 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
+
+;; Request StackIR explicitly. This optimizes.
+;; RUN: wasm-opt %s --generate-stack-ir --optimize-stack-ir -all               --print-stack-ir | filecheck %s --check-prefix=REQUESTED
+
+;; As above, but disallow it later. This does not optimize.
+;; RUN: wasm-opt %s --generate-stack-ir --optimize-stack-ir -all --no-stack-ir --print-stack-ir | filecheck %s --check-prefix=DISALLOWD
+
+;; As above, but flip it, so we allow it after disallowing. This optimizes.
+;; RUN: wasm-opt %s --no-stack-ir --generate-stack-ir --optimize-stack-ir -all --print-stack-ir | filecheck %s --check-prefix=REALLOWED
```

Comment:
Can you explain how these commands end up with different results? I am baffled.

Reply:
* `--no-stack-ir` says "disable it"
* `--optimize-stack-ir` says "enable it"

And the last one on the commandline wins, just like `-fno-exceptions -fexceptions` etc.

Reply:
Oh, I see, `--optimize-stack-ir` is handled in `ToolOptions` 👍